### PR TITLE
Pull Requests: Create swagger spec

### LINKED ISSUE #8167: Pull Requests MVP: Create swagger spec
Reference: https://github.com/treeverse/lakeFS/issues/8164

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #8167

## Change Description

Create swagger spec, generated code and controller stubs for pull requests

### Background

[Epic](https://github.com/treeverse/lakeFS/issues/8166)

### CODE REVIEW

**Comment by N-o-Z** [comment]
Actual changed files:
- swagger.yaml
- controller

All the rest are auto generated code

File: pkg/api/controller.go [hunk 1]
```diff
@@ -5176,6 +5176,26 @@ func (c *Controller) PostStatsEvents(w http.ResponseWriter, r *http.Request, bod
...
 
+func (c *Controller) ListPullRequests(w http.ResponseWriter, r *http.Request, repository string, params apigen.ListPullRequestsParams) {
+	writeResponse(w, r, http.StatusNotImplemented, nil)
+}
+
+func (c *Controller) CreatePullRequest(w http.ResponseWriter, r *http.Request, body apigen.CreatePullRequestJSONRequestBody, repository string) {
+	writeResponse(w, r, http.StatusNotImplemented, nil)
+}
+
+func (c *Controller) DeletePullRequest(w http.ResponseWriter, r *http.Request, repository string, pullRequest string) {
```
Comment (itaigilo) [inline pkg/api/controller.go:5187]: Should be `repositoryID` and `pullRequestID`.
Also relevant for `GetPullRequest` and `UpdatePullRequest`.

File: api/swagger.yml [hunk 2]
```diff
@@ -5582,6 +5653,159 @@ paths:
...
+          description: too many requests
+        default:
+          $ref: "#/components/responses/ServerError"
+
+  /repositories/{repository}/pulls/{pull_request}:
+    parameters:
+      - in: path
+        name: repository
+        required: true
+        schema:
```
Comment (itaigilo) [inline api/swagger.yml:5735]: what about description?

File: api/swagger.yml [hunk 3]
```diff
@@ -5582,6 +5653,159 @@ paths:
...
+          $ref: "#/components/responses/Unauthorized"
+        403:
+          $ref: "#/components/responses/Forbidden"
+        404:
+          $ref: "#/components/responses/NotFound"
+        420:
+          description: too many requests
+        default:
+          $ref: "#/components/responses/ServerError"
+    patch:
```
Comment (itaigilo) [inline api/swagger.yml:5783]: Note that in the Design [Issue](https://github.com/treeverse/lakeFS/issues/8164) it's designed different - 
But what's written in the PR is far better (hence I think the design should be updated).

**Review by itaigilo — APPROVED** [review]
LGTM,
See minor comments.

It'll probably require some changes soon once I'll start implementing the UI, but we'll work on this as we go.

File: pkg/api/controller.go (hunk 1 above)
Comment (N-o-Z) [reply pkg/api/controller.go:5187]: For repository our convention in all other functions is `repository`
I'll update `pullRequest`

**Pull request merged**